### PR TITLE
Fix fullstack definition

### PR DESCRIPTION
**Pull request by jehna**
### Fix ambiguous reference between node.js and "full stack"

Full stack means all layers of the stack (e.g. LAMP stack stands for Linux, Apache, Mysql, PHP). If we use node.js on backend and frontend, it still means we'll need to use something (e.g. Linux) to run node.js. Nowadays it's common to also virtualize the operating system by using e.g. Docker as an additional layer in the stack.

### Remove sentence about fullstack's relation to JS

According to Google Search Trends, the term "full stack" [was indeed introduced (or at least popularized) after node.js was born][1].

However emphasizing the connection/causation between JavaScript and the term "full stack" is irrelevant in this context, since the term "full stack" covers more than just frontend and backend.

[1]: https://trends.google.com/trends/explore?date=all&q=%22full%20stack%22,%22node.js%22

### Add clarification about OS being part of stack
    
> For example, to develop a web application the architect defines the stack as the target operating system, web server, database, and programming language.

https://en.wikipedia.org/wiki/Solution_stack

### CODE REVIEW

File: osa0.md [hunk 1]
```diff
@@ -697,9 +697,9 @@ Käytännössä kaikki websovellukset sisältävät (ainakin) kaksi "kerrosta",
...
+Tämän kurssin kontekstissa full stack -sovelluskehitys tarkoittaa sitä, että fokus on kaikissa sovelluksen osissa, niin frontendissä kuin backendissä sekä taustalla olevassa tietokannassa. Myös palvelimen käyttöjärjestelmä lasketaan usein osaksi stackia.
 
-Ohjelmoimme myös palvelinpuolta, eli backendia Javascriptilla, käyttäen [Node.js](https://nodejs.org/en/)-suoritusympäristöä. Näin full stack -sovelluskehitys saa vielä uuden ulottuvuuden, käytämme samaa kieltä pinon kaikissa osissa. Full stack -sovelluskehitys ei välttämättä edellytä sitä, että kaikissa "sovelluspinon" kerroksissa on käytössä sama kieli (Javascript). Termi "full stack" on kuitenkin (todennäköisesti) lanseerattu vasta sen jälkeen kun Node.js mahdollisti Javascriptin käyttämisen kaikkialla.
+Ohjelmoimme myös palvelinpuolta, eli backendia Javascriptilla, käyttäen [Node.js](https://nodejs.org/en/)-suoritusympäristöä. Näin full stack -sovelluskehitys saa vielä uuden ulottuvuuden, kun voimme käyttää samaa kieltä pinon useammassa kerrokessa. Full stack -sovelluskehitys ei välttämättä edellytä sitä, että kaikissa "sovelluspinon" kerroksissa on käytössä sama kieli (Javascript).
```
Comment (JuhQ) [inline osa0.md:702]: `kerrokessa`

File: osa0.md [hunk 2]
```diff
@@ -697,9 +697,9 @@ Käytännössä kaikki websovellukset sisältävät (ainakin) kaksi "kerrosta",
...
 
-Tämän kurssin kontekstissa full stack -sovelluskehitys tarkoittaa sitä, että fokus on kaikissa sovelluksen osissa, niin frontendissä kuin backendissä sekä taustalla olevassa tietokannassa.
+Tämän kurssin kontekstissa full stack -sovelluskehitys tarkoittaa sitä, että fokus on kaikissa sovelluksen osissa, niin frontendissä kuin backendissä sekä taustalla olevassa tietokannassa. Myös palvelimen käyttöjärjestelmä lasketaan usein osaksi stackia.
 
-Ohjelmoimme myös palvelinpuolta, eli backendia Javascriptilla, käyttäen [Node.js](https://nodejs.org/en/)-suoritusympäristöä. Näin full stack -sovelluskehitys saa vielä uuden ulottuvuuden, käytämme samaa kieltä pinon kaikissa osissa. Full stack -sovelluskehitys ei välttämättä edellytä sitä, että kaikissa "sovelluspinon" kerroksissa on käytössä sama kieli (Javascript). Termi "full stack" on kuitenkin (todennäköisesti) lanseerattu vasta sen jälkeen kun Node.js mahdollisti Javascriptin käyttämisen kaikkialla.
```
Comment (JuhQ) [inline osa0.md:702]: Miksi termin määritys poistettu?

**Review by JuhQ — CHANGES_REQUESTED** [review]

File: osa0.md (hunk 2 above)
Comment (jehna) [reply osa0.md:702]: Tämä lause yhdisti mielenkiintoisesti  JavaScriptin sekä "fullstack"-termin  toisiinsa syy-seuraussuhteeseen; toisin sanottuna lauseen voisi ymmärtää esimerkiksi näin:

> Node.js mahdollisti Javascriptin käyttämisen kaikkialla, jota kutsutaan termillä "full stack"

"Full stack" on kuitenkin laajempi käsite kuin pelkästään JavaScript-kehittäminen, joten tämän yhteyden tekeminen kuulosti omituiselta.

Comment (JuhQ) [reply osa0.md:702]: Roger!

**Review by JuhQ — APPROVED** [review]

File: osa0.md (hunk 1 above)
Comment (jehna) [reply osa0.md:702]: Korjattu commitissa c17d6b2

**Pull request merged**